lib/public/components/Filters/common/filters/timeRangeFilter.js

```diff
@@ -0,0 +1,83 @@
+/**
+ *  @license
+ *  Copyright CERN and copyright holders of ALICE O2. This software is
+ *  distributed under the terms of the GNU General Public License v3 (GPL
+ *  Version 3), copied verbatim in the file "COPYING".
+ *
+ *  See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ *  In applying this license CERN does not waive the privileges and immunities
+ *  granted to it by virtue of its status as an Intergovernmental Organization
+ *  or submit itself to any jurisdiction.
+ */
+import { dropdown } from '../../../common/popover/dropdown.js';
+import { h } from '/js/src/index.js';
+import { dateTimeInput } from '../../../common/form/inputs/dateTimeInput.js';
+import { getLocaleDateAndTime } from '../../../../utilities/dateUtils.js';
+
+/**
+ * Format the current timestamp range display
+ *
+ * @param {Partial<Period>} the current period
+ * @return {vnode|string} the resulting component
+ */
+const formatTimeRange = ({ from, to }) => {
+    let parts = [];
+
+    if (from === null && to === null) {
+        return '-';
+    }
+
+    // eslint-disable-next-line require-jsdoc
+    const formatTimestamp = (timestamp) => {
+        if (timestamp === undefined) {
+            return '-';
+        }
+
+        const { date, time } = getLocaleDateAndTime(timestamp);
+        return h('.flex-column.g1', [h('.f6', date), h('', time)]);
+    };
+
+    // eslint-disable-next-line require-jsdoc
+    const label = (content) => h('.gray-darker', content);
+
+    if (from === undefined && to === undefined) {
+        parts = '-';
+    } else if (from === undefined) {
+        parts = [label('Before'), formatTimestamp(to)];
+    } else if (to === undefined) {
+        parts = [label('After'), formatTimestamp(from)];
+    } else {
+        parts = [label('From'), formatTimestamp(from), label('to'), formatTimestamp(to)];
+    }
```

Comment:
I think it can be achieved much shorter (if you don't care about zeroes): 
```
const parts = [
        ... from && [label('After'), formatTimestamp(from) || []],
        ... to && [label('Before'), formatTimestamp(to) || []],
];
```

Reply:
another think is taking care about null values

Reply:
With your solution, we are missing the case where both are defined.
This can be handled with a ternary inside what you proposed, but I think that this would make the code unclear, don't you think?

Something like:
```js
const parts = [
        ... from !== undefined && [label(to !== undefined ? 'From' : 'After'), formatTimestamp(from)],
        ... to !== undefined && [label(from !== undefined ? 'to' : 'Before'), formatTimestamp(to)],
];
```

Reply:
ok, I missed that labels differ. If altering them is important for some reason (e.g. it's nicer) then let it be as is